packages/@aws-cdk/aws-cloudfront/lib/response-headers-policy.ts

```diff
@@ -140,6 +140,14 @@ export class ResponseHeadersPolicy extends Resource implements IResponseHeadersP
       xssProtection: behavior.xssProtection,
     };
   }
+
+  private generateUniqueId(): string {
+    const name = Names.uniqueId(this);
+    if (name.length > 128) {
```

Comment:
Even though it appears you can successfully create up to 129 characters, I used 128 because that seemed more standard.